src/tty/mod.rs

```diff
@@ -0,0 +1,45 @@
+// Copyright 2016 Joe Wilm, The Alacritty Project Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//! tty related functionality
+
+use mio;
+use std::io;
+
+#[cfg(not(windows))]
+mod unix;
+#[cfg(not(windows))]
+pub use self::unix::*;
+
+#[cfg(windows)]
+mod windows;
+#[cfg(windows)]
+pub use self::windows::*;
+
+/// This trait defines the behaviour needed to read and/or write to a stream.
+/// It defines an abstraction over mio's interface in order to allow either one
+/// read/write object or a seperate read and write object.
+pub trait EventedRW {
```

Comment:
The name for this trait is a bit odd. Maybe `Event` would be better than `Evented` here? I'm not 100% what exactly is meant by 'evented'. Also `ReadWrite` is short enough to write it out completely imho. I think `EventReadWrite` would be a bit closer to Rust's guidelines for trait names.

Reply:
`Evented` refers to the fact that it implements `mio::event::Evented`, so I think it makes sense to maintain that?

Changing `RW` to `ReadWrite` makes sense though.

Reply:
Oh okay, I wasn't aware that `Evented` is a trait from `mio`. Should be fine to use `EventedReadWrite`.